buffer/src/test/java/io/netty/buffer/SlicedByteBufTest.java

```diff
@@ -227,4 +231,33 @@ public void testWriteIso88591CharSequenceExpand() {
     public void testWriteUtf16CharSequenceExpand() {
         super.testWriteUtf16CharSequenceExpand();
     }
+
+    @Test
+    public void ensureWritableWithEnoughSpaceShouldNotThrow() {
+        ByteBuf buf = Unpooled.buffer(10);
```

Comment:
I didn't want to be subject to the size manipulation in the `newBuffer` method and I would have to unwrap the buffer to adjust underlying indexes ... however I can do this incase we use a different allocator in the future.